handler/core/src/integrationTest/java/io/knotx/fragments/handler/integration/ConsumerAcceptsInvalidFragmentTest.java

```diff
@@ -0,0 +1,226 @@
+    Fragment fragment = fragment("snippet", "failing-composite-node-task",
+        "<div>some content</div>");
+
+    verifyDebugData(configFile, fragment, HTML_BODY_TO_EXECUTION_LOG, COMPOSITE_ASSERTIONS, vertx,
+        testContext);
+  }
+
+  @Test
+  @DisplayName("Expect invalid JSON fragment with single node task contains debug data.")
+  void invalidJsonFragmentWithSingleNodeTask(Vertx vertx, VertxTestContext testContext)
+      throws Throwable {
+    String configFile = "conf/invalid-single-node-task-with-consumer.conf";
+    Fragment fragment = fragment("json", "failing-single-node-task", "{}");
+    Consumer<FragmentExecutionLog> assertions = executionLog -> {
+      GraphNodeExecutionLog graphRootNode = executionLog.getGraph();
+      checkTimelines(graphRootNode);
+      checkMissingNode(graphRootNode);
+    };
+
+    verifyDebugData(configFile, fragment, JSON_BODY_TO_EXECUTION_LOG, assertions, vertx,
+        testContext);
+  }
+
+  @Test
+  @DisplayName("Expect invalid JSON fragment with composite node task contains debug data.")
+  void invalidJsonFragmentWithCompositeNodeTask(Vertx vertx, VertxTestContext testContext)
+      throws Throwable {
+    String configFile = "conf/invalid-composite-node-task-with-consumer.conf";
+    Fragment fragment = fragment("json", "failing-composite-node-task", "{}");
+
+    verifyDebugData(configFile, fragment, JSON_BODY_TO_EXECUTION_LOG, COMPOSITE_ASSERTIONS, vertx,
+        testContext);
+  }
+
+  private static final Consumer<FragmentExecutionLog> COMPOSITE_ASSERTIONS =
+      executionLog -> {
+        GraphNodeExecutionLog graphRootNode = executionLog.getGraph();
+
+        assertEquals(2, graphRootNode.getSubtasks().size());
+        GraphNodeExecutionLog successSubtask = graphRootNode.getSubtask("success-action")
+            .orElseThrow(() -> new AssertionError("Subtask node not found!"));
+        checkTimelines(successSubtask);
+        assertEquals(LoggedNodeStatus.SUCCESS, successSubtask.getStatus());
+
+        GraphNodeExecutionLog failingSubtask = graphRootNode.getSubtask("failing-action")
+            .orElseThrow(() -> new AssertionError("Subtask node not found!"));
+        checkTimelines(failingSubtask);
+        assertEquals(LoggedNodeStatus.ERROR, failingSubtask.getStatus());
+
+        checkMissingNode(graphRootNode);
+      };
+
+
+  private static final Function<String, FragmentExecutionLog> JSON_BODY_TO_EXECUTION_LOG = fragmentBody -> {
+    assertTrue(new JsonObject(fragmentBody).containsKey("_knotx_fragment"));
+    JsonObject debugData = new JsonObject(fragmentBody).getJsonObject("_knotx_fragment");
+    return new FragmentExecutionLog(debugData);
+  };
+
+  private static final Function<String, FragmentExecutionLog> HTML_BODY_TO_EXECUTION_LOG = fragmentBody -> {
+    String scriptRegexp =
+        "<script data-knotx-debug=\"log\" data-knotx-id=\".*\" type=\"application/json\">(?<fragmentEventJson>.*?)</script>";
```

Comment:
It seems that we're using this regex in some places, maybe it's a good idea to export it to the dedicated constant in order to easier modify it in the future if needed?